### PR TITLE
Orchestrating Cloud Events Meetup Blog Post

### PR DESCRIPTION
**Pull request by salaboy**
This is a blog post Pull Request. 
This blog post describes the demo presented at the Knative meetup linked in the content.

### CODE REVIEW

File: blog/articles/orchestrating-cloud-events-with.md
Comment (mpetason) [inline blog/articles/orchestrating-cloud-events-with.md:23]:
```suggestion
Until this point, this is just a normal Knative application, since you are using Knative Eventing which natively supports Cloud Events, you can integrate more tools to the ecosystem as you will see in the next section.
```

Maybe "since" instead of and, because. It reads a little bit better, but I can also see where the "and" would read alright.

**Comment by mpetason** [comment]
The content is great. I'm going to need help with correct punctuation/modifying some of the run on sentences. The message gets across well, just some of the sentences are a bit long and can probably be broken up to provider more compact bits of information.

**Comment by salaboy** [comment]
Hi Mike, sure.. I am @salaboy in slack, feel free to drop me some message there if you are ok with doing it asynchronously.

Cheers

File: blog/articles/orchestrating-cloud-events-with.md
Comment (salaboy) [reply blog/articles/orchestrating-cloud-events-with.md:23]: @mpetason up to you.. really.. both work fine for me..

**Comment by salaboy** [comment]
@mpetason any way that I can help you with the corrections?

**Comment by mpetason** [comment]
@salaboy we're going to need to wait for a review from someone who is better with technical writing.

**Comment by salaboy** [comment]
@mpetason do you know how does the process works? is someone going to pick this up at some point or should we try to find someone who can review this?

**Comment by mpetason** [comment]
@salaboy you can post this in the slack channel to get real time updates, however it's going to wait until someone else has a chunk of time available to go through and make edits.

File: blog/articles/orchestrating-cloud-events-with.md
Comment (csantanapr) [inline blog/articles/orchestrating-cloud-events-with.md:1]: Blog posts need hugo metadata at the start of the file. See one of the other blog posts

Here are the guidelines for blog posts https://github.com/knative/docs/tree/master/blog#blog-guidelines

Comment (lionelvillard) [reply blog/articles/orchestrating-cloud-events-with.md:1]: maybe instead of `Cloud Events`, write `CloudEvents`

File: blog/articles/orchestrating-cloud-events-with.md
Comment (lionelvillard) [inline blog/articles/orchestrating-cloud-events-with.md:19]: nit: Cloud-Native -> Cloud Native

File: blog/articles/orchestrating-cloud-events-with.md
Comment (lionelvillard) [inline blog/articles/orchestrating-cloud-events-with.md:97]: I suppose you mean "push events **when** workflow models ...", not if.

File: blog/articles/orchestrating-cloud-events-with.md
Comment (salaboy) [reply blog/articles/orchestrating-cloud-events-with.md]: @lionelvillard done! thanks for the suggestion

File: blog/articles/orchestrating-cloud-events-with.md
Comment (salaboy) [reply blog/articles/orchestrating-cloud-events-with.md]: @lionelvillard done!

File: blog/articles/orchestrating-cloud-events-with.md
Comment (salaboy) [reply blog/articles/orchestrating-cloud-events-with.md]: @lionelvillard ok.. I've rephrased it.. thanks.. I think that it is much clear now.

**Comment by salaboy** [comment]
@lionelvillard @csantanapr all changes requested done and pushed .. let me know if you think I should change something else. Cheers

**Comment by lionelvillard** [comment]
thanks @salaboy, looks great, both the blog and the demo!  
/lgtm

**Comment by salaboy** [comment]
@lionelvillard thanks a lot!

File: blog/articles/orchestrating-cloud-events-with.md
Comment (macruzbar) [inline blog/articles/orchestrating-cloud-events-with.md:10]:
```suggestion
A couple of weeks ago, I presented at the **Knative Meetup** ([Video](https://www.youtube.com/watch?v=msDDdqmyEFA&list=PLQjzPfIiEQLIEpoCPxBYAVrjqy6LxLtQ8), [Slides](https://www.slideshare.net/salaboy/orchestrating-cloud-events-knative-meetup-2020)) about how you can leverage the Cloud Native workflow engine [Zeebe](http://zeebe.io) to understand, enhance and orchestrate your applications that are already using [CloudEvents](https://cloudevents.io).
```

File: blog/articles/orchestrating-cloud-events-with.md
Comment (macruzbar) [inline blog/articles/orchestrating-cloud-events-with.md:58]:
```suggestion
Now when the customer reaches the **Payment Sent** state, meaning that they have already made a reservation, the model waits for the **Payment Sent** CloudEvent to arrive.
```

File: blog/articles/orchestrating-cloud-events-with.md
Comment (macruzbar) [inline blog/articles/orchestrating-cloud-events-with.md:62]:
```suggestion
As you can see above, two timer boundary events have been associated with the **Payment Sent** state. The bottom one, with dashed lines, is a non-interrupting event, meaning that it will not disrupt the actual flow, it will be just triggered in a fire-and-forget fashion. This results in the Payment Reminder being sent and, because it can be configured as a recurring timer, it will trigger every X amount of time (for the presentation this was set to 10 seconds).
```

File: blog/articles/orchestrating-cloud-events-with.md
Comment (macruzbar) [inline blog/articles/orchestrating-cloud-events-with.md:68]:
```suggestion
The top one, with solid lines, is an interrupting event, meaning that it will cause the normal flow of the model to be discarded and continue only to the steps defined from the timer (in this case, **Reservation Time Out**). For this scenario, the interrupting timer is set to be triggered after X amount of time after the tickets' reservation was made (for the presentation, this was set to 2 minutes). After the reservation times out, the website redirects the customer back to the beginning.
```

File: blog/articles/orchestrating-cloud-events-with.md
Comment (macruzbar) [inline blog/articles/orchestrating-cloud-events-with.md:70]:
```suggestion
For both timers, you get full traceability (e.g. when and how often they were triggered). You can see these details graphically in **Operate**:
```

File: blog/articles/orchestrating-cloud-events-with.md
Comment (macruzbar) [inline blog/articles/orchestrating-cloud-events-with.md:74]:
```suggestion
You can see that the non-interrupting timer was triggered several times, generating payment reminders in the frontend. Even if we don’t implement the frontend notifications, you can use these timers to measure and classify how much time it actually takes your customer to make the payment. This can be really useful to classify different categories; for example, how much time it takes, on average, for people to provide payment details.
```

File: blog/articles/orchestrating-cloud-events-with.md
Comment (macruzbar) [inline blog/articles/orchestrating-cloud-events-with.md:80]:
```suggestion
The **Payment Reminder** and **Reservation Time Out** status, shown in the model above, represent when we cross the line from simply listening to events to actually emitting CloudEvents from a workflow model. Now the workflow model is not only helping you to visualize and understand what your Cloud Native applications are doing but driving and interacting with your services.
```

Comment (macruzbar) [inline blog/articles/orchestrating-cloud-events-with.md:80]:
```suggestion
The **Payment Reminder** and **Reservation Time Out** states, shown in the model below, represent when we cross the line from simply listening to events to actually emitting CloudEvents from a workflow model. Now the workflow model is not only helping you to visualize and understand what your Cloud Native applications are doing but driving and interacting with your services.
```

Comment (macruzbar) [reply blog/articles/orchestrating-cloud-events-with.md:80]: adding as a separate revision because I am not sure if you want to commit this change. I noticed you can see these status in both the model above, and the one right below this paragraph, so up to you.

File: blog/articles/orchestrating-cloud-events-with.md
Comment (macruzbar) [inline blog/articles/orchestrating-cloud-events-with.md:90]:
```suggestion
In the following screenshot, inside Camunda Operate, you can quickly visualize how many instances of the workflows are at a given stage as well as how many workflows have finished, and in which state they finished:
```

File: blog/articles/orchestrating-cloud-events-with.md
Comment (macruzbar) [inline blog/articles/orchestrating-cloud-events-with.md:94]:
```suggestion
Another advantage of using a workflow engine is flow control. By using flow control elements such as Exclusive Gateways, you can delegate some high-level decisions (usually encoding business logic) to the workflow engine:
```

Comment (macruzbar) [reply blog/articles/orchestrating-cloud-events-with.md:94]: capitalize every time, or don't capitalize.

File: blog/articles/orchestrating-cloud-events-with.md
Comment (macruzbar) [inline blog/articles/orchestrating-cloud-events-with.md:191]:
```suggestion
Watch this demo live here: 
{{< youtube msDDdqmyEFA >}}
```

**Review by macruzbar — CHANGES_REQUESTED** [review]
Thank you for working on this blogpost, @salaboy ! I have requested a few changes; after these, I'll add the lgtm label. Thanks!

File: blog/articles/orchestrating-cloud-events-with.md
Comment (salaboy) [reply blog/articles/orchestrating-cloud-events-with.md]: I am ok with the one above

**Comment by salaboy** [comment]
@macruzbar done :) !

**Comment by salaboy** [comment]
@macruzbar hmm.. there seems to be some test broken .. not related to this article
```
=== FAIL: test/e2e TestSampleApp/kotlin (211.09s)
1720
    sampleapptestbase.go:81: Prepare source project
1721
    sampleapptestbase.go:110: Pushing docker image to: 'gcr.io/knative-boskos-49/docs-e2e-img/3154/helloworld-kotlin:latest'
1722
    sampleapptestbase.go:112: Error building docker image: Sending build context to Docker daemon  9.728kB

1723
        Step 1/7 : FROM gradle:4.10 as builder
1724
        4.10: Pulling from library/gradle
skipped 108 lines unfold_more
1833
        
1834
        * Get more help at https://help.gradle.org
      
1835
        
1836
        BUILD FAILED in 51s
1837
        The command '/bin/sh -c gradle clean build --no-daemon' returned a non-zero code: 1
1838
    --- FAIL: TestSampleApp/kotlin (211.09s)
1839
1840
=== FAIL: test/e2e TestSampleApp (0.01s)

```

**Pull request merged**
